happi/device.py

```diff
@@ -348,6 +346,12 @@ def __iter__(self):
     def __len__(self):
         return len(self.post())
 
+    def __copy__(self):
+        return type(self)(**dict(self))
+
+    def __deepcopy__(self, memo):
+        return self.__copy__()
```

Comment:
I'm rusty on this, are you overriding some default behavior in the `__deepcopy__`? Or is there no default behavior for either method using the other and we must always supply both a `__copy__` and a `__deepcopy__` to enable copying?

Reply:
I don't believe `__deepcopy__` calls `__copy__` by default, so it's up to us to implement.

It's possible this 'deep copy = shallow copy' as I've implemented it may come back to bite us for mutable objects contained within the device. We could wait to figure that out or consider this "good enough for now" - up to you guys.

Reply:
The alternative is to call deepcopy recursively on the dict items, right? I don't think it matters that much, but maybe it's better to have no deepcopy at all than one that lies to us?

Reply:
Quite reasonable. It was dumb of me not to implement it properly

Reply:
not dumb... just what you could get as good enough for now and that makes sense! 😉

Reply:
Thanks, guys. That's what these reviews are about after all! Thinking about it a bit more, maybe it's not as bad as I thought.

How's 39edf8c look? It flips this around such that `__copy__` and `__deepcopy__` always return a deep copy with the help of the `copy` module. 

A shallow copy when you ask for a deepcopy is without question bad. A deep copy is safer in all regards, resulting in less confusion for users.

Reply:
So now copy returns a deepcopy as well? At #39edf8c I am also confused because `memo` is not used... should you add the copy to `memo`? I am not familiar with implementing __deepcopy__ sorry if dumb question.

Reply:
I'm also not savvy on the `memo` argument but this implementation looks good to me.

Reply:
`memo` is for keeping track of making the deepcopy, so you can stash stuff that newly-created sub-objects might need to be copied.

It should be fine as-is, since we just have a [potentially nested] dictionary of built-in types in the form of the device information.